### PR TITLE
Integrate the new reserve REST API endpoint #265 

### PR DESCRIPTION
**Pull request by ines-cruz**
Merge and release after https://github.com/inveniosoftware/invenio-rdm-records/pull/560

### CODE REVIEW

File: src/lib/state/actions/deposit.js [hunk 1]
```diff
@@ -74,3 +75,56 @@ export const discard = (event, formik) => {
...
+  return async (dispatch, getState, config) => {
+    const controller = config.controller;
+
+    dispatch({
+      type: RESERVE_PID_STARTED,
+    });
+
+    // PIDS-FIXME: this is not the latest version of the form values,
+    // it will replace the form values with what was in the record
+    const draft = getState().deposit.record;
```
Comment (ntarocco) [inline src/lib/state/actions/deposit.js:94]: @zzacharo I have no idea how to solve this. Using `submitFormData` instead will not wait for the state to be updated. We can chat about this is you have ideas.

File: src/lib/DepositApiClient.js [hunk 2]
```diff
@@ -167,4 +167,48 @@ export class DepositApiClient {
...
+   * Calls the API to reserve a PID.
+   *
+   */
+  async reservePID(links, pidType) {
+    return this.createResponse(() => {
+      const link = `${links.self}/pids/${pidType}`;
+      // const link = _get(links, `self_${pidType}`, '');
+      // if (link === '') {
+      //   throw Error(`Cannot get the link to discard the PID for ${pidType}`);
+      // }
```
Comment (ppanero) [inline src/lib/DepositApiClient.js:181]:
```suggestion
```

File: src/lib/DepositApiClient.js [hunk 3]
```diff
@@ -167,4 +167,48 @@ export class DepositApiClient {
...
+   * Calls the API to discard a previously reserved PID.
+   *
+   */
+  async discardPID(links, pidType) {
+    return this.createResponse(() => {
+      const link = `${links.self}/pids/${pidType}`;
+      // const link = _get(links, `self_${pidType}`, '');
+      // if (link === '') {
+      //   throw Error(`Cannot get the link to discard the PID for ${pidType}`);
+      // }
```
Comment (ppanero) [inline src/lib/DepositApiClient.js:203]:
```suggestion
```

File: src/lib/DepositController.js [hunk 4]
```diff
@@ -220,4 +224,54 @@ export class DepositController {
...
+   */
+  reservePID = async (draft, pidType, { formik, store }) => {
+    const recordSerializer = store.config.recordSerializer;
+
+    const response = await this.apiClient.reservePID(draft, pidType);
+
+    let data = recordSerializer.deserialize(response.data || {});
+    let errors = recordSerializer.deserializeErrors(response.errors || []);
+
+    if (200 <= response.code && response.code < 300 && _isEmpty(errors)) {
```
Comment (ppanero) [inline src/lib/DepositController.js:239]: doubt: arent we just returning 200? why is the range needed?

File: src/lib/components/Identifiers/PIDField.js [hunk 5]
```diff
@@ -44,61 +38,57 @@ class ReservePIDBtn extends Component {
...
     const { onManagedUnmanagedChange } = this.props;
     const isManagedSelected = value === 'managed' ? true : false;
-    this.setState({ isManagedSelected: isManagedSelected });
     onManagedUnmanagedChange(isManagedSelected);
   };
 
   render() {
-    const { isManagedSelected } = this.state;
-    const { pidLabel } = this.props;
+    // const { isManagedSelected } = this.state;
```
Comment (ppanero) [inline src/lib/components/Identifiers/PIDField.js:90]:
```suggestion
```

File: src/lib/DepositController.js (hunk 4 above)
Comment (ntarocco) [reply src/lib/DepositController.js:239]: Honestly, this should very much refactored, it is a copy/paste as above to keep things the same and refactor all of them at once in the right way.

File: src/lib/components/Identifiers/PIDField.js (hunk 5 above)
Comment (ntarocco) [reply src/lib/components/Identifiers/PIDField.js:91]: 👍🏼

**Pull request merged**